File: Leet_Code-1160/Leet_Code-1160.py
'''
Question 1160
'''

class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        ans = 0
        for w in words:
            wc = 0
            n = len(w)
            for c in w:
                if chars.count(c) >= w.count(c):
                    wc += 1
            if wc == n:
                ans += wc
        return ans

File: Leet_Code-1160/Leet_Code-1160.java
/*
 * Question 1160
 */

class Solution {
    public int countCharacters(String[] words, String chars) {
        int ans = 0;
        for (String w: words){
            int[] charCount = new int[26];
            boolean isGoodWord = true;
            for (char c: chars.toCharArray()){
                charCount[c - 'a']++;
            }
            for (char c: w.toCharArray()){
                if (charCount[c - 'a'] == 0){
                    isGoodWord = false;
                    break;
                }
                charCount[c - 'a']--;
            }
            if (isGoodWord){
                ans += w.length();
            }
        }
        return ans;
    }
}
